components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java

```diff
@@ -0,0 +1,291 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.secret.management.v1.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.api.server.common.error.APIError;
+import org.wso2.carbon.identity.api.server.common.error.ErrorResponse;
+import org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants;
+import org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementServiceHolder;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.SecretAddRequest;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.SecretResponse;
+import org.wso2.carbon.identity.api.server.secret.management.v1.model.SecretUpdateRequest;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementClientException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementServerException;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secrets;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants.CONFIG_MGT_ERROR_CODE_DELIMITER;
+import static org.wso2.carbon.identity.api.server.secret.management.common.SecretManagementConstants.ERROR_PREFIX;
+
+/**
+ * Invoke internal OSGi service to perform secret management operations.
+ */
+public class SecretManagementService {
+
+    private static final Log log = LogFactory.getLog(SecretManagementService.class);
+
+    /**
+     * Create a secret.
+     *
+     * @param secretAddRequest secret post request.
+     * @return secret.
+     */
+    public SecretResponse addSecret(String secretType, SecretAddRequest secretAddRequest) {
+
+        validateSecretAddRequest(secretAddRequest);
+        Secret requestDTO, responseDTO;
+        try {
+            requestDTO = buildSecretRequestDTOFromSecretAddRequest(secretAddRequest);
+            responseDTO = SecretManagementServiceHolder.getSecretConfigManager().addSecret(secretType, requestDTO);
+        } catch (SecretManagementException e) {
+            throw handleSecretMgtException(e, SecretManagementConstants.ErrorMessage.ERROR_CODE_ERROR_ADDING_SECRET,
+                    secretAddRequest.getName());
+        }
+        return buildSecretResponseFromResponseDTO(responseDTO);
+    }
+
+    /**
+     * To create Secret Response object for the post request
+     *
+     * @param secretReq secret object.
+     * @return {@link SecretResponse} .
+     */
+    private SecretResponse buildSecretResponseFromResponseDTO(Secret secretReq) {
```

Comment:
shouldn't the variable name be `secretResponse` ?

Reply:
Addressed with commit [3dd3011](https://github.com/wso2/identity-api-server/pull/315/commits/3dd301183d2884eead3721c8c43a1fe33f3c2052)